packages/jarvis/src/install.js

```diff
@@ -1,114 +1,52 @@
-
-    return new Promise((resolve, reject) => {
-      if (error) {
-        return reject(error)
-      }
-
-      const dependencies = modules.reduce(
-        (allDependencies, moduleDependencies) => [...allDependencies, ...moduleDependencies],
-        [],
-      )
-
-      if (dependencies.length) {
-        this.spinner.text = 'Installing dependencies'
-        return resolve(new Promise((addResolve, addReject) => {
-          exec(`yarn add ${dependencies.join(' ')}`, (installError) => {
-            if (installError) {
-              return addReject(new Error('Unable to install dependencies'))
-            }
-
-            return addResolve()
-          })
-        }))
-      }
-
-      return resolve()
-    })
-  }
-
-  async buildPackageJson() {
-    this.spinner.text = 'Attempting to read existing package.json'
-    const packageJsonPath = path.join(process.cwd(), 'package.json')
-    const [readError, initialPackageJsonRaw] = await task(readFile(packageJsonPath))
-
-    // eslint-disable-next-line consistent-return
-    return new Promise((resolve, reject) => {
-      try {
-        // If we had a read error, just create a basic package.json file
-        // or else try and parse the found package.json file
-        const initialPackageJson = readError ? { private: true } : JSON.parse(initialPackageJsonRaw)
-
-        // Run the package.json through the config transform
-        this.spinner.text = 'Applying package config transforms'
-        const packageJson = this.configQueue.reduce(
-          (obj, { key, transform }) => ({
-            ...obj,
-            [key]: transform(obj[key]),
-          }),
-          initialPackageJson,
-        )
-
-        fs.writeFile(packageJsonPath, JSON.stringify(packageJson, null, 2), (writeError) => {
-          if (writeError) {
-            return reject(new Error('Could not write package.json'))
-          }
-
-          return resolve()
-        })
-      } catch (parseError) {
-        return reject(new Error('Could not parse package.json'))
-      }
-    })
-  }
-
-  execute() {
-    this.spinner.start()
-
-    this.buildPackageJson()
-      .then(() => {
-        this.installPackages()
-          .then(() => {
-            this.spinner.succeed('Done!')
-          })
-          .catch(() => {
-            this.spinner.fail('Failed installing packages')
-          })
-      })
-      .catch(() => {
-        this.spinner.fail('Failed buidling package.json')
-      })
+const chalk = require('chalk')
+
+const { log } = require('./util/console')
+const prompt = require('./util/prompt')
+const resolveSerial = require('./util/resolveSerial')
+
+const filesTask = require('./tasks/files')
+const dependenciesTask = require('./tasks/dependencies')
+const configChangesTask = require('./tasks/configChanges')
+
+// Export the addItem functions from the tasks
+exports.addConfigChange = configChangesTask.addItem
```

Comment:
Just wondering why we're using require & exports over import & export?

Reply:
Last time I check ECMAScript modules are experimental in Node so you have to use Babel or similar to transpile to common JS. This project isn't yet using Babel for Node.